### PR TITLE
Add support for JEDEC ID and SFDP

### PR DESCRIPTION
**Pull request by osenft**
This PR builds on top of https://github.com/google/tock-on-titan/pull/184.

```
----------------------
`make prtest` summary:
----------------------
git rev-parse HEAD
b75f21934e4855c88f32fae3ab49559ea93a00f6
git status
On branch jedec
nothing to commit, working tree clean
```

### CODE REVIEW

**Comment by osenft** [comment]
I just rebased this PR to #184. PTAL.

File: kernel/h1/src/spi_device.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@ use kernel::ReturnCode;
 use spiutils::protocol::flash::AddressMode;
 use spiutils::protocol::flash::OpCode;
 
+fn get_data_byte_or_default(data: &[u8], idx: usize, default: u8) -> u8 {
+    if idx < data.len() { data[idx] } else { default }
```
Comment (jrvanwhy) [inline kernel/h1/src/spi_device.rs]: This can be `data.get(idx).unwrap_or(default)`

File: userspace/otpilot/src/main.rs [hunk 2]
```diff
@@ -160,73 +163,113 @@ impl rsa::Builder for NoRsa {
...
+    Format(core::fmt::Error),
+}
+
+impl From<FromWireError> for SpiProcessorError {
+    fn from(err: FromWireError) -> Self {
+        SpiProcessorError::FromWire(err)
+    }
+}
+
+impl From<core::num::TryFromIntError> for SpiProcessorError {
```
Comment (jrvanwhy) [inline userspace/otpilot/src/main.rs]: I'm not sure I like the use of `From<core::num::TryFromIntError>` to create a `SpiProcessorError::FromWire`. It doesn't feel like a clean abstraction. As far as I can tell, it is only used in one location.

Perhaps we should remove this `From` implementation and replace `u16::try_from(tx_cursor.consumed_len())?` with `u16::try_from(tx_cursor.consumed_len()).map_err(|_| FromWireError::OutOfRange.into())?`

File: userspace/otpilot/src/sfdp.rs [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+
+    for idx in 0..sfdp.len() {
+        data[idx] = sfdp[idx];
+    }
+    for idx in sfdp.len()..data.len() {
+        data[idx] = !0;
+    }
+
+    Ok(())
+}
```
Comment (jrvanwhy) [inline userspace/otpilot/src/sfdp.rs]: Nitpick: the last line of this file is missing an EOL character, please add one.

**Review by jrvanwhy — CHANGES_REQUESTED** [review]

File: kernel/h1/src/spi_device.rs (hunk 1 above)
Comment (osenft) [reply kernel/h1/src/spi_device.rs]: Ha, yeah, I did not know that, thank you! I tried to actually merge the get_data_byte_or_default back into the code, but the resulting line was very hard to read due to references and dereferencing and the resulting need of parantheses. So I just updated the helper function and added a comment.

File: userspace/otpilot/src/main.rs (hunk 2 above)
Comment (osenft) [reply userspace/otpilot/src/main.rs]: Good point. The other cases should be clean. Thanks for pointing this out!

File: userspace/otpilot/src/sfdp.rs (hunk 3 above)
Comment (osenft) [reply userspace/otpilot/src/sfdp.rs]: One day I'll figure out how to make Sublime do that automatically ... sigh.

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**